### ISSUE TITLE
.NET 7 Preview 4 install breaks RC3 install experience

### ISSUE THREAD

**Issue by dcuccia** [dotnet/maui#7422]
### Description

Hi!

I was excited to try out .Net MAUI, so I installed Visual Studio 2022 17.3.0 Preview 1.1 on my Windows 11 machine, and this is what I see:

![image](https://user-images.githubusercontent.com/479617/169902513-0bbf2c20-9368-4517-9481-dedf889e3c5b.png)

Can't restore ios or android packages - could that be part of the problem?

Here are all of my .NETs:

![image](https://user-images.githubusercontent.com/479617/169902685-3fe8af1e-97df-4acc-a9e8-f6a5d24de173.png)

Workloads were no help:

![image](https://user-images.githubusercontent.com/479617/169903289-b435b4f3-e645-4275-b38e-9f723ba43c17.png)

Here's what I see from `dotnet --info`:

```
.NET SDK:
 Version:   7.0.100-preview.4.22252.9
 Commit:    6bc31786a8

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.100-preview.4.22252.9\

global.json file:
  Not found

Host:
  Version:      7.0.0-preview.4.22229.4
  Architecture: x64
  Commit:       9c37a3b3eb

.NET SDKs installed:
  3.1.419 [C:\Program Files\dotnet\sdk]
  5.0.408 [C:\Program Files\dotnet\sdk]
  6.0.101 [C:\Program Files\dotnet\sdk]
  6.0.105 [C:\Program Files\dotnet\sdk]
  6.0.201 [C:\Program Files\dotnet\sdk]
  6.0.203 [C:\Program Files\dotnet\sdk]
  6.0.300 [C:\Program Files\dotnet\sdk]
  7.0.100-preview.4.22252.9 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.25 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0-preview.4.22251.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.22 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.23 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.25 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0-preview.4.22229.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.22 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.23 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.25 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0-preview.4.22229.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Download .NET:
  https://aka.ms/dotnet-download

Learn about .NET Runtimes and SDKs:
  https://aka.ms/dotnet/runtimes-sdk-info
```

Here's the installation info from the About... dialog:

```
Microsoft Visual Studio Professional 2022
Version 17.3.0 Preview 1.1
VisualStudio.17.Preview/17.3.0-pre.1.1+32519.111
Microsoft .NET Framework
Version 4.8.04161

Installed Version: Professional

.NET Core Debugging with WSL   1.0
.NET Core Debugging with WSL

ADL Tools Service Provider   1.0
This package contains services used by Data Lake tools

ASA Service Provider   1.0

ASP.NET and Web Tools 2019   17.3.122.33185
ASP.NET and Web Tools 2019

Azure App Service Tools v3.0.0   17.3.122.33185
Azure App Service Tools v3.0.0

Azure Data Lake Tools for Visual Studio   2.6.5000.0
Microsoft Azure Data Lake Tools for Visual Studio

Azure Functions and Web Jobs Tools   17.3.122.33185
Azure Functions and Web Jobs Tools

Azure Stream Analytics Tools for Visual Studio   2.6.5000.0
Microsoft Azure Stream Analytics Tools for Visual Studio

C# Tools   4.3.0-1.22254.1+9919d7e7bd753404a5d2328e5e3fb2de635169f3
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.2.6 (master@34d6af2)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Microsoft Azure Hive Query Language Service   2.6.5000.0
Language service for Hive query

Microsoft Azure Stream Analytics Language Service   2.6.5000.0
Language service for Azure Stream Analytics

Microsoft Azure Tools for Visual Studio   2.9
Support for Azure Cloud Services projects

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Mono Debugging for Visual Studio   17.3.11 (4f73250)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   6.3.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Razor (ASP.NET Core)   17.0.0.2222701+751db1ebea5e6a9ecc7fa57fe447180422afa610
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   17.0.62204.01010
Microsoft SQL Server Data Tools

ToolWindowHostedEditor   1.0
Hosting json editor into a tool window

TypeScript Tools   17.0.10420.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.3.0-1.22254.1+9919d7e7bd753404a5d2328e5e3fb2de635169f3
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.1.0-beta.22178.3+6da0245a7ce4bb8483b8d1f2993c8ecaea967ad9
Microsoft Visual F# Tools

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

VSPackage Extension   1.0
VSPackage Visual Studio Extension Detailed Info

Xamarin   17.3.0.151 (d17-3-p1.1@5a59bfa)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   17.3.0.33 (remotes/origin/d17-3-1@9c0d11a29)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin.Android SDK   12.3.99.58 (main/64b22fc)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: adf1bc4
    Java.Interop: xamarin/java.interop/release/6.0.3xx@2a882d2d
    ProGuard: Guardsquare/proguard/v7.0.1@912d149
    SQLite: xamarin/sqlite/3.38.2@7b1e016
    Xamarin.Android Tools: xamarin/xamarin-android-tools/main@fc3c2ac
```

What can I do next to troubleshoot?

### Steps to Reproduce

1. Install Visual Studio 2022 17.3.0 Preview 1.1
2. File->New Project MAUI.
3. (uninstall 1 + restart, try 1 again...same issue)

### Version with bug

Release Candidate 3 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

I was *not* able test on other platforms

### Affected platform versions

Unclear - won't build on Windows

### Did you find any workaround?

_No_

**Comment by adegeo** [comment]
Hi!

Did you select the MAUI workload in the VS installer?

Perhaps the workload search with SDK 7 preview doesn't have those workloads published? I don't really know how that works. You could also force downgrade the `dotnet` command temporarily just to test by adding a `global.json` file in the same directory you're running the workload command:

```json
{
  "sdk": {
    "version": "6.0.300"
  }
}
```

Also, try using `dotnet workload search maui` here are my results

![image](https://user-images.githubusercontent.com/67293991/169905739-283aa726-a272-4b58-b6a9-e716ad7a88d6.png)

**Comment by dcuccia** [comment]
@adegeo yes, but something's clearly broken - workload search not returning anything:

![image](https://user-images.githubusercontent.com/479617/169908117-346851c9-0f3f-4392-b45e-9b807f562436.png)

Great tip on global.json! Did that and now...

```
➜ dotnet workload list

Installed Workload Ids      Installation Source
-----------------------------------------------
maui-windows                VS 17.3.32519.111
maui-maccatalyst            VS 17.3.32519.111
maccatalyst                 VS 17.3.32519.111
maui-ios                    VS 17.3.32519.111
ios                         VS 17.3.32519.111
maui-android                VS 17.3.32519.111
android                     VS 17.3.32519.111

Use `dotnet workload search` to find additional workloads to install.
```

`dotnet new maui`, spin up VS Preview, and bingo-bango:

![image](https://user-images.githubusercontent.com/479617/169909475-38680907-5d02-48ad-beb4-93dab2475a65.png)

**Comment by dcuccia** [comment]
So, is this a corrupted .NET 7 Preview 4 install, or the expected behavior for SxS with .NET 7 Preview 4? Answer probably informs on how many BUILD attendees might fall into the same trap tomorrow.

**Comment by adegeo** [comment]
Yeah, I'll send an email to the team to get this on their radar ASAP.

**Comment by dcuccia** [comment]
I renamed the ticket to ".NET 7 Preview 4 install breaks RC3 install experience"

**Comment by adegeo** [comment]
@dcuccia can you tell me, did you install .net 7 before, during, or after upgrading VS?

**Comment by dcuccia** [comment]
I'm sorry, but I'm not 100% sure in this case, I generally tend to install the latest Visual Studio Preview, and then grab the latest .NET SDK preview from the dot.net website. But, I've been dabbling with NET 7 Preview 4 filters for Minimal API in a project I've been working on in Rider. I can say that when troubleshooting today, I tried to wipe VS22 Preview, restart, and reinstall with the MAUI workload selected, without touching a stand-alone preview installer and still observed this behavior.

Possibly this was redundant this time around, and perhaps that caused shenanigans?

**Comment by mattjohnsonpint** [comment]
FWIW, I believe this is the known issue mentioned here:
https://github.com/dotnet/maui/wiki/Known-Issues#net-maui-templates-do-not-appear-in-visual-studio

**Comment by adegeo** [comment]
That sounds right!

**Comment by dcuccia** [comment]
@mattjohnsonpint thank you - yes that's it.

**Comment by adegeo** [comment]
@dcuccia do you remember how you created the project? Were the templates visible in visual studio at the time?

**Comment by dcuccia** [comment]
@adegeo I recall not seeing the templates at first upon upgrading to the latest VS 22 Preview. I think I then attempted a repair of the x-plat mobile workload with the VS Installer UI, and then perhaps they showed up (but weren't compiling)? I then uninstalled and reinstalled the preview and still saw the templates but had the same issue, until the global.json suggestion here. Sorry, struggling to re-trace my steps.

This is definitely a great use-case for GitHub Dev Box - keep preview bits on the remote instance, production bits local. I've had enough success with .Net/VS previews lately that I've become maybe a bit overconfident in installing the latest-greatest SxS with the daily driver.
